adam/curriculum_to_html.py

```diff
@@ -0,0 +1,177 @@
+from pathlib import Path
+from typing import Iterable
+
+from attr import attrs
+from vistautils.parameters import Parameters
+
+from adam.curriculum.phase1_curriculum import GAILA_PHASE_1_CURRICULUM
+from adam.experiment import InstanceGroup
+from adam.language.dependency import LinearizedDependencyTree
+from adam.perception import PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+)
+from adam.situation.high_level_semantics_situation import HighLevelSemanticsSituation
+
+
+@attrs(frozen=True, slots=True)
+class CurriculumToHtmlDumper:
+    """
+    Class to turn an `InstanceGroup` into an html document
+    """
+
+    def dump_to_html(
+        self,
+        instance_groups: Iterable[
+            InstanceGroup[
+                HighLevelSemanticsSituation,
+                LinearizedDependencyTree,
+                DevelopmentalPrimitivePerceptionFrame,
+            ]
+        ],
+        *,
+        output_destination: Path,
+        title: str = "Instance Group",
+    ):
+        r"""
+        Method to take a list of `InstanceGroup`\ s and turns each one into an individual page
+        
+        Given a list of InstanceGroups and an output directory of *outputdestination* along with 
+        a *title* for the pages the generator loops through each group and calls the internal 
+        method to create HTML pages.
+        """
+        for (idx, instance_group) in enumerate(instance_groups):
+            CurriculumToHtmlDumper._dump_instance_group(
+                instance=instance_group,
+                output_destination=output_destination.joinpath(f"{title}{idx}.html"),
+                title=f"{title} {idx}",
+            )
+
+    @staticmethod
```

Comment:
Un-static-ify all the methods. Apart from static factory methods, use static methods sparingly.